### PR TITLE
Release 0.23.2 backports part I

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: pandas/tests/series/test_analytics.py [hunk 1]
```diff
@@ -1866,6 +1866,189 @@ def s_main_dtypes():
     return df
 
 
+class TestMode(object):
```
Comment (jorisvandenbossche) [inline pandas/tests/series/test_analytics.py]: the addition of this class is probably something that went wrong in fixing the conflict. Will see what the tests do, if this is was simply moved in another PR and the tests passes, would leave it like this here.

Comment (TomAugspurger) [reply pandas/tests/series/test_analytics.py]: I thought I had an issue with a mode test in https://github.com/pandas-dev/pandas/pull/21442/files, but didn't see it in a quick scan. Could be misremembering.
